pkg/addons/addons.go

```diff
@@ -394,24 +405,53 @@ func Start(wg *sync.WaitGroup, cc *config.ClusterConfig, toEnable map[string]boo
 
 	var awg sync.WaitGroup
 
-	defer func() { // making it show after verifications( not perfect till #7613 is closed)
+	deferredAddons := []string{}
+	enabledAddons := []string{}
+
+	defer func() { // making it show after verifications (see #7613)
 		register.Reg.SetStep(register.EnablingAddons)
-		out.T(style.AddonEnable, "Enabled addons: {{.addons}}", out.V{"addons": strings.Join(toEnableList, ", ")})
+		out.T(style.AddonEnable, "Enabled addons: {{.addons}}", out.V{"addons": enabledAddons})
 	}()
+	var addonErr error
 	for _, a := range toEnableList {
+		// Run the gcp-auth addon last to make sure everything else is up
```

Comment:
This is still a race condition, as addon enablement does not block until the pods are available.

I think you can remove this special case by naming a `failurePolicy` for the webhook instead, so that it fails open if unavailable:

https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#failure-policy

Reply:
For an example test case: `/out/minikube start --addons=dashboard --addons=efk --addons=freshpod --addons=istio --addons=istio-provisioner --addons=olm --memory=16384 --driver=hyperkit`

That said, it is a race condition, and may not be easy to trigger. Better to just fail open.

Reply:
I think adding the failurePolicy is a good idea just in general, but this code isn't supposed to eliminate the race, just reduce it as much as possible. In my testing it seems to be very effectively at precisely that.

Reply:
Now that failurePolicy is in place, can this ordering code be safely removed?

The nice thing about failurePolicy is that it also eliminates the race condition when you use `minikube addons enable` while another pod is scheduling.

Reply:
so just in my empirical testing, the race condition still happens fairly often without this code, which makes me think I don't quite understand what's happening here.